### PR TITLE
express-notes-api by jaden bertino

### PR DESCRIPTION
**Pull request by jadenbertino**
In the photos, some of the IDs are a string instead of a number, but I fixed this !!

# GET
image

# GET /:id
Success
image

ID doesn't exist
image

ID is not a positive integer
image

# POST
Success
image

No content property in request body
image

Internal Server Error (writing to derp/data.json)
image

# DELETE
Success
image

ID doesn't exist
image

ID not a positive integer
image

Internal Server Error (writing to derp/data.json)
image

# PUT
Success
image

ID not a positive integer
image

ID doesn't exist
image

No content provided
image

Internal Server Error (writing to derp/data.json)

### CODE REVIEW

**Comment by jadenbertino** [comment]
I sincerely apologize for the atrocious commit names 😭

**Comment by jadenbertino** [comment]
just realized I need to fix the id that gets attached to newly created notes / updated notes

File: express-notes-api/server.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import express from 'express';
+import { getFileContent, updateDatabase } from './fileUtils.js';
+import { validateContent, validateId } from './validationUtils.js';
+
+const app = express();
+const PORT = 8080;
+const NOTES_URL = '/api/notes';
+const data = JSON.parse(await getFileContent('data.json'));
+const { notes } = data;
```
Comment (TimelikeClosure) [inline express-notes-api/server.js]: While reading the data once at startup saves on request time, it also tends to result in stale data, particularly when multiple clients are making requests or multiple instances of web servers are using a shared data source. Instead, read a fresh copy of the data source with each request.

File: express-notes-api/fileUtils.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    if (!file) throw new TypeError('writeToFile called with undefined "file" argument');
+    if (!textContent) throw new TypeError('writeToFile called with undefined "textContent" argument');
+
+    const textToWrite = new Uint8Array(Buffer.from(textContent));
+    await writeFile(file, textToWrite);
+
+  } catch (err) {
+    console.error(err);
+  }
+}
```
Comment (TimelikeClosure) [inline express-notes-api/fileUtils.js]: File system read / write failures during request time should result in `500 Internal Server Error` responses:

---
image

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]
@jadenbertino Please add screenshots verifying that the failure cases behave as expected.

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**